.github/workflows/llama-perf.yml

```diff
@@ -1,140 +1,276 @@
+    strategy:
+      fail-fast: false
+      max-parallel: 1
+      matrix:
+        jax-version: ["0.6.0", "0.7.1"]
+        model-name: ["train_moe", "train_dense_8b"]
+        include:
+          - jax-version: "0.6.0"
+            jaxlib-version: "0.6.0"
+            model-name: "train_moe"
+            docker-image: "rocm/jax:rocm7.0.2-jax0.6.0-py3.12-ubu24"
+          - jax-version: "0.6.0"
+            jaxlib-version: "0.6.0"
+            model-name: "train_dense_8b"
+            docker-image: "rocm/jax:rocm7.0.2-jax0.6.0-py3.12-ubu24"
+          - jax-version: "0.7.1"
+            jaxlib-version: "0.7.1"
+            model-name: "train_moe"
+            docker-image: "ghcr.io/rocm/jax-ubu24.rocm702:a2dc0f639643e3e4d03088b77114527b59c377c1" # TODO: tag nightly
+          - jax-version: "0.7.1"
+            jaxlib-version: "0.7.1"
+            model-name: "train_dense_8b"
+            docker-image: "ghcr.io/rocm/jax-ubu24.rocm702:a2dc0f639643e3e4d03088b77114527b59c377c1" # TODO: tag nightly
+    steps:
       - name: Checkout source repo
         uses: actions/checkout@v4
         with:
           repository: AMD-AGI/scif_repro
           token: ${{ secrets.SECRET_TOKEN }}
-          path: llama-perf
-      - name: Move wheels
+      - name: Download JAX wheels
+        # TODO: update for dwn last nightly whl
+        env:
+          GH_TOKEN: ${{ secrets.SECRET_TOKEN }}
         run: |
-          mkdir -p ${{ github.workspace }}/llama-perf/wheelhouse
-          mv wheelhouse/*.whl ${{ github.workspace }}/llama-perf/wheelhouse
+          if [ "${{ matrix.jax-version }}" == "0.6.0" ]; then
+            mkdir -p wheelhouse
+            REL_URL="https://github.com/ROCm/rocm-jax/releases/download/rocm-jax-v0.6.0"
+            curl -L --output-dir wheelhouse -O \
+            "$REL_URL/jax_rocm7_pjrt-0.6.0-py3-none-manylinux_2_28_x86_64.whl"
+            curl -L --output-dir wheelhouse -O \
+            "$REL_URL/jax_rocm7_plugin-0.6.0-cp312-cp312-manylinux_2_28_x86_64.whl"
+          else
+            gh run download 18648819000 -R ROCm/rocm-jax -n plugin_wheels_r7.0.0 -D wheelhouse
+            rm -v wheelhouse/*311*
+          fi
+      - name: Download TE wheel artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: te-wheel-jax${{ matrix.jax-version }}-${{ needs.build-te-wheels.outputs.architecture }}
+          path: wheelhouse
       - name: Print wheels
         run: |
-          ls -la ${{ github.workspace }}/llama-perf/wheelhouse
+          ls -la ${{ github.workspace }}/wheelhouse
       - name: Disable non-root check
         run: |
-          sed -i '/python.toolchain(/a\    ignore_root_user_error = True,' \
-          "${{ github.workspace }}/llama-perf/MODULE.bazel"
+          sed -i '/python.toolchain(/a\    ignore_root_user_error = True,' MODULE.bazel
       - name: Patch source files
+        # these patches will be removed when the repo is updated/tested accordingly
```

Comment:
Which repo? Right now this whole step is pretty unreadable, so it'd be good to eliminate if possible